portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java

```diff
@@ -31,20 +33,58 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-
+import java.text.*;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/delete-data")
 public class DeleteDataServlet extends HttpServlet {
-
+  
   @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Comment").setKeysOnly();
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment");
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    PreparedQuery results = datastore.prepare(query);    
+    PreparedQuery results = datastore.prepare(query);
+    String stringPostTime = (String) request.getParameter("postTime");
+    DateFormat format = new SimpleDateFormat("MMM d, yyyy h:mm:ss a");
+    Date postTime;
+    try {
+      postTime = format.parse(stringPostTime);
+    } catch (Exception e) {
+      System.out.println(e);
+      postTime = new Date();
+    }
+    UserService userService = UserServiceFactory.getUserService();
+    String userId = userService.getCurrentUser().getUserId();
+    String isAdmin = String.valueOf(userService.isUserAdmin());
```

Comment:
Good approach for checking credentials on the server side.  This definitely would be less likely to hack compared to checking for this on the client side.